pkg/controller/master/upgradelog/controller.go

```diff
@@ -146,7 +145,12 @@ func (h *handler) OnUpgradeLogChange(_ string, upgradeLog *harvesterv1.UpgradeLo
 			return nil, err
 		}
 		// The creation of the Logging resource will indirectly bring up fluent-bit DaemonSet and fluentd StatefulSet
-		if _, err := h.loggingClient.Create(prepareLogging(upgradeLog)); err != nil && !apierrors.IsAlreadyExists(err) {
+		url := fmt.Sprintf("http://%s.%s/%s", util.HarvesterClusterRepoSvcName, util.CattleSystemNamespaceName, rancherLoggingChartValuesPath)
```

Comment:
The default images in `values.yaml` may be overwritten by pre-defined values from `managedchart` /  `addons`, and if the users modify the values in `managedchart` / `addon`, that will be the final values to be used.

Seems we need to also pick values from `managedchart` /  `addons` to overwrite the default ones as following.

```
images:
  config_reloader:
    repository: rancher/mirrored-jimmidyson-configmap-reload
    tag: v0.4.0
  fluentbit:
    repository: rancher/mirrored-fluent-fluent-bit
    tag: 1.9.3
  nodeagent_fluentbit:
    os: "windows"
    repository: rancher/fluent-bit
    tag: 1.9.3
  fluentbit_debug:
    repository: rancher/mirrored-fluent-fluent-bit
    tag: 1.9.3-debug
  fluentd:
    repository: rancher/mirrored-banzaicloud-fluentd
    tag: v1.14.6-alpine-5
```

Reply:
Thanks for the heads-up. Now the implementation becomes fetching image info from `apps.catalog.cattle.io/v1` and consolidates the default values with the user-defined values.

Reply:
cc @chrisho, now it's get from apps, which contains the final value of a chart.